formats/gltf/writer.go

```diff
@@ -317,6 +325,142 @@ func rgbToFloatArr(c color.Color) [3]float64 {
+	for _, m := range scene.Models {
+		if err := w.AddMesh(m); err != nil {
+			return fmt.Errorf("failed to add mesh %q: %w", m.Name, err)
+		}
+	}
+
+	// Add lights
+	for _, light := range scene.Lights {
+		w.AddLight(light)
+	}
+
+	return nil
+}
+
+func (w *Writer) addSceneWithInstancing(scene PolyformScene) error {
+	// Map to track unique models and their mesh indices
+	modelIndices := make(map[*PolyformModel]int)
+
+	// First add all unique models
+	for _, model := range scene.Models {
+		// Add mesh and record its index
+		meshIndex := len(w.meshes)
+		modelIndices[&model] = meshIndex
+
+		// Create the mesh - process geometry, materials etc
+		primitiveAttributes := make(map[string]int)
+
+		for _, val := range model.Mesh.Float4Attributes() {
+			primitiveAttributes[polyformToGLTFAttribute(val)] = len(w.accessors)
+			w.WriteVector4(attributeType(val), model.Mesh.Float4Attribute(val))
+		}
+
+		for _, val := range model.Mesh.Float3Attributes() {
+			primitiveAttributes[polyformToGLTFAttribute(val)] = len(w.accessors)
+			w.WriteVector3(attributeType(val), model.Mesh.Float3Attribute(val))
+		}
+
+		for _, val := range model.Mesh.Float2Attributes() {
+			primitiveAttributes[polyformToGLTFAttribute(val)] = len(w.accessors)
+			w.WriteVector2(attributeType(val), model.Mesh.Float2Attribute(val))
+		}
+
+		indicesIndex := len(w.accessors)
+		w.WriteIndices(model.Mesh.Indices(), model.Mesh.AttributeLength())
+
+		var materialIndex *int
+		var err error
+		if model.Material != nil {
+			materialIndex, err = w.AddMaterial(model.Material)
+			if err != nil {
+				return fmt.Errorf("failed to add material %q: %w", model.Material.Name, err)
+			}
+		}
+
+		var mode *PrimitiveMode = nil
+		if model.Mesh.Topology() == modeling.PointTopology {
+			p := PrimitiveMode_POINTS
+			mode = &p
+		}
+
+		w.meshes = append(w.meshes, Mesh{
+			ChildOfRootProperty: ChildOfRootProperty{
+				Name: model.Name,
+			},
+			Primitives: []Primitive{
+				{
+					Indices:    &indicesIndex,
+					Attributes: primitiveAttributes,
+					Material:   materialIndex,
+					Mode:       mode,
+				},
+			},
+		})
+
+		// Handle any skeleton/animation data
+		if model.Skeleton != nil {
+			var skinIndex *int
+			skinIndex, _ = w.AddSkin(*model.Skeleton)
+			nodeIndex := len(w.nodes)
+			w.nodes = append(w.nodes, Node{
+				Mesh: &meshIndex,
+				Skin: skinIndex,
+			})
+			w.scene = append(w.scene, nodeIndex)
+
+			if len(model.Animations) > 0 {
+				w.AddAnimations(model.Animations, *model.Skeleton, nodeIndex)
+			}
+		}
+	}
+
+	// Then create nodes for all instances
+	for _, instance := range scene.Instances {
+		meshIndex := modelIndices[instance.Model]
+
+		// Create node referencing the mesh with instance transform
+		nodeIndex := len(w.nodes)
+		newNode := Node{Mesh: &meshIndex}
+
+		if instance.Translation != nil {
+			arr := [3]float64{instance.Translation.X(), instance.Translation.Y(), instance.Translation.Z()}
```

Comment:
For some reason `vector3.Number.ToArr()` returns a slice `[]float64` (in this case), rather than actual Go fixed-length array `[3]float64`. I would prefer to change the `ToArr() []T` signature to `ToArr() [3]T`, that will simpify this bit a little, and be more true to what "to array" implies. But that will be a change of the public API, so not sure how acceptable that would be.

Reply:
Yeah, unfortunately, the vector lib isn't in beta, so I'm less comfortable changing signatures

Added `ToFixedArr()` in [v1.8.0](https://github.com/EliCDavis/vector/releases/tag/v1.8.0)